snuba/cli/dlq_manager.py

```diff
@@ -0,0 +1,151 @@
+import os
+from typing import MutableSequence, Optional, Sequence, Tuple
+
+import click
+from arroyo import Message, Partition
+from arroyo import Topic as ArroyoTopic
+from arroyo.backends.kafka import KafkaConsumer, KafkaPayload
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.factory import STORAGES_WITH_DLQ
+from snuba.utils.streams.configuration_builder import build_kafka_consumer_configuration
+
+
+@click.group()
+def dlq_manager() -> None:
+    """
+    Tools for Snuba's Dead Letter Queues
+    """
+    pass
+
+
+@dlq_manager.command()
+@click.option(
+    "--storage",
+    "storage_set",
+    type=click.Choice(
+        [storage_set_key.value for storage_set_key in STORAGES_WITH_DLQ.keys()]
+    ),
+    help="The storage set to list dead letters for",
+    required=True,
```

Comment:
A `storage` and a `storage set` are two quite distinct things in Snuba. Which one is this supposed to be?

Reply:
I do mean storage set but I'm not entirely sure which it should be, DLQ's are defined per storage but I think storages under the same storage set share the same dead-letter topic.

Reply:
Personally I think it should be defined at the storage level. Not all storages in a storage set might have a DLQ. That does not mean that storages can never share a DLQ, just that they are defined that the storage level and are not the same by default.

Reply:
If the DLQ is shared across the storages in a set, it would be weird to choose from a list of storages and not storage sets for the CLI. Choosing any storage under the same set would show the same results when querying for invalid messages.

Could we ever have different storages under the same storage set using different DLQ topics?

Reply:
If we look at this line...
```
STORAGE_SETS_WITH_DLQ = {
    storage.get_storage_set_key().value: topic
    for storage, topic in __STORAGE_TO_DLQ_TOPIC.items()
    if topic is not None
}
```
...we see that DLQ topic is defined at the storage level, and if there are 2 storages with different DLQ topics, then the first one will be clobbered by the second.

I think it's also possible multiple storage sets will end up sharing the same DLQ, in which case it might be equally weird to pick a storage set and see items from multiple storage sets in there.

If --storage or --storage-set is just a roundabout way of specifying a topic anyway, why don't you just pass the topic via the CLI instead?